API/Extensions/ExceptionHandlingMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Net;
+using System.Text.Json;
+
+namespace API.Extensions;
+
+public class ExceptionHandlingMiddleware(RequestDelegate next)
+{
+    public async Task InvokeAsync(HttpContext httpContext)
+    {
+        try
+        {
+            await next(httpContext);
+        }
+        catch (Exception ex)
+        {
+            await HandleExceptionAsync(httpContext, ex);
+        }
+    }
+
+    private static async Task HandleExceptionAsync(HttpContext context, Exception ex)
+    {
+        var responseMessage = ex switch
+        {
+            _ => new
+            {
+                Status = HttpStatusCode.InternalServerError,
+                ex.Message
+            }
+        };
+
+        var response = JsonSerializer.Serialize(responseMessage);
```

Comment:
Do we have an async overload like SerializeAsync?

Reply:
![image](https://github.com/Gh0stF1ghter/Project-Rent-n-Ride/assets/72450306/f7e0265f-f298-4774-b794-c71a7921c22b)
Only in .net 9)